engine/function/src/templates/Numeric.ftl

```diff
@@ -2294,6 +2496,37 @@ public class Numeric {
      * @param weights weights
      * @return weighted sum of non-null values.
      */
+   <#if pt.valueType.isInteger && pt2.valueType.isInteger >
+    public static long wsum(${pt.vector} values, ${pt2.vector} weights) {
+        if (values == null || weights == null) {
+            return NULL_LONG;
+        }
+
+        final long n = values.size();
+
+        if (n != weights.size()) {
+            throw new IllegalArgumentException("Incompatible input sizes: " + values.size() + ", " + weights.size());
+        }
+
+        long vsum = 0;
+
+        try (
+            final ${pt.vectorIterator} vi = values.iterator();
+            final ${pt2.vectorIterator} wi = weights.iterator()
+        ) {
+            while (vi.hasNext()) {
+                final ${pt.primitive} c = vi.${pt.iteratorNext}();
+                final ${pt2.primitive} w = wi.${pt2.iteratorNext}();
+
+                if (!isNull(c) && !isNull(w)) {
+                    vsum += c * w;
```

Comment:
Java question: is `c * w` protected from overflow? Do we need to cast one of these to `long` so we can prevent an unintentional overflow?